clients/src/main/java/org/apache/kafka/common/record/KafkaLZ4BlockInputStream.java

```diff
@@ -114,16 +139,17 @@ private void readHeader() throws IOException {
      *
      * @throws IOException
      */
-    private void readBlock() throws IOException {
+    private void readBlock() throws IOException, LZ4Exception {
         int blockSize = Utils.readUnsignedIntLE(in);
 
         // Check for EndMark
         if (blockSize == 0) {
             finished = true;
-            // TODO implement content checksum, update flg.validate()
+            if (this.flg.isContentChecksumSet())
```

Comment:
I quite prefer to be explicit w/ `this` when referencing object state variables. but I'll change